source/administration/legacy-upgrade.md

```diff
@@ -1,17 +1,109 @@
+      - Note: If public links are enabled, upgrading from `v3.3.x` and earlier to `v3.4.x` and later will invalidate existing public links due to a security upgrade allowing admins to invalidate links by resetting a public link salt from the System Console.
+      - Note: RHEL6 and Ubuntu installations must verify the line `limit nofile 50000 50000` is included in `/etc/init/mattermost.conf` file. See the [installation guide](https://docs.mattermost.com/guides/administrator.html#install-guides) for your operating system for more details.
+      - Note: RHEL7 and Ubuntu installations must verify the line `LimitNOFILE=49152` is included in the `/etc/systemd/system/mattermost.service` file. See the [installation guide](https://docs.mattermost.com/guides/administrator.html#install-guides) for your operating system for more details.
+      1. Download [the latest version of the Mattermost Server](https://about.mattermost.com/download/). In the following command, replace `X.X.X` with the version that you want to download:
+          - **Enterprise Edition**: `wget https://releases.mattermost.com/X.X.X/mattermost-X.X.X-linux-amd64.tar.gz`
+      2. Review past and upcoming deprecation notices [listed on our website](https://about.mattermost.com/deprecated-features/).
+2. Stop the Mattermost Server
+      1. Consider posting an announcement to active teams about stopping the Mattermost server for an upgrade.
+      2. To stop the server run `sudo stop mattermost`.
+3. Backup your data
+      1. Back up your `config.json` file, which contains your system configuration. This will be used to restore your current settings after the new version is installed.
+      2. Backup your database using your organization's standard procedures for backing up MySQL or PostgreSQL.
+      3. If you're using local file storage, back up the location where files are stored.
+4. Install new version
+      1. Run `tar -xvzf mattermost-X.X.X-linux-amd64.tar.gz` to decompress the upgraded version and replace the current version of Mattermost on disk, where `X.X.X` is the version number to which you are upgrading.  
+5. Restore the state of your server
+      1. Copy the backed up version of `config.json` in place of the default `config.json`.
+6. Start your server and address any setting changes relevant in the latest version of Mattermost
+      1. Run `sudo start mattermost`.
+      2. Opening the **System Console** and saving a change will upgrade your `config.json` schema to the latest version using default values for any new settings added.
+7. If you have TLS set up on your Mattermost server, you must activate the CAP_NET_BIND_SERVICE capability to allow the new Mattermost binary to bind to low ports.
+
+      1. ``cd {install-path}``
+      2. ``sudo setcap CAP_NET_BIND_SERVICE=+ep ./bin/platform``
+
+After the server is upgraded, users might need to refresh their browsers to experience any new features.
+
+## Upgrade Team Edition to 3.0.3
```

Comment:
It's been that way for quite some time... is a common enough scenario to be concerned about?